.github/workflows/pre-release.yml

```diff
@@ -18,47 +18,34 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Setup node
-        uses: actions/setup-node@v2
-        with:
-          node-version: '16'
-          cache: 'npm'
-          registry-url: https://registry.npmjs.org
+      - name: Setup node and install deps
+        uses: ./.github/workflows/actions/node-setup
 
       - name: Config git user
         run: |
           git config --global user.name "${{ github.actor }}"
           git config --global user.email "${{ github.actor }}@users.noreply.github.com"
           git fetch --depth=1 origin +refs/tags/*:refs/tags/*
 
-      - name: Cache Node Modules
-        id: cache-node
-        uses: actions/cache@v2
-        with:
-          path: node_modules
-          key: ${{ runner.OS }}-build-${{ hashFiles('**/package.json') }}
-
-      - name: Install Node Dependencies
-        if: steps.cache-node.outputs.cache-hit != 'true'
-        run: npm ci --legacy-peer-deps
-        env:
-          CI: TRUE
-
       - name: Bump versions (pre-release)
-        run: npx lerna version --yes --conventional-prerelease --no-changelog --no-git-tag-version --preid ${{ github.event.inputs.tag }}
+        run: |
+          pnpm changeset pre enter next
```

Comment:
read about [pre release](https://github.com/changesets/changesets/blob/main/docs/prereleases.md)
this is to advance to use within a pipeline, use [snapshot](https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md)